core/templates/pages/topic-viewer-page/practice-tab/practice-tab.component.ts

```diff
@@ -100,11 +103,11 @@ export class PracticeTabComponent implements OnInit {
   }
 
   checkIfQuestionsExist(subtopicIndices: boolean[]): void {
-    const skillIds = [];
+    const skillIds: string[] = [];
     this.questionsStatusCallIsComplete = false;
     for (let idx in subtopicIndices) {
       if (subtopicIndices[idx]) {
-        skillIds.push(this.availableSubtopics[idx].getSkillIds());
+        skillIds.push(...this.availableSubtopics[idx].getSkillIds());
```

Comment:
Hm, have you tested this functionality? It looks like the logic has changed a bit here, not sure..

Reply:
Here ```skillIds``` is of type ```string[]```. Also ```this.availableSubtopics[idx].getSkillIds()``` will return an array of string. If I remove ```...```, I get an error: ```Argument of type 'string[]' is not assignable to parameter of type 'string'.ts(2345)```

Reply:
Thanks for explaining but I would recommend that you also test the functionality to make sure nothing is broken.

Reply:
Ah, looks like `skillIds` used to be a singleton array of strings and the only place where it gets used later is here:

https://github.com/oppia/oppia/blob/d156fd023370b1de57d99985cb73bf3d15398d11/core/templates/domain/question/question-backend-api.service.ts#L93-L95

The `join` method flattens the array of arrays and joins the strings so it didn't cause any errors. Thanks for checking, I don't think any further action is required here.

Reply:
Thanks. That's really insightful. Btw, How did you follow skillIds till this file? (Did you use some kind of search?)